### ISSUE TITLE
Rename Name-Shifted Profiling Interface (PMPI) to Interception Interface

### ISSUE THREAD

**Issue by wesbland** [mpi-forum/mpi-issues#584]
# Problem

As part of developing the proposal for QMPI, we needed to differentiate between the existing interface and the newly proposed interface. This pointed out the strange existing name.

PMPI is used for more than just profiling and it would be best if the name reflected that.

# Proposal

Replace “name-shifted profiling interface”, “profiling interface”, etc. (or similar text) anywhere it is used to say “name-shifted interception interface”.

# Changes to the Text

https://github.com/mpi-forum/mpi-standard/pull/724

# Impact on Implementations

None

# Impact on Users

Might break PDF searches

# References and Pull Requests

**Comment by mahermanns** [comment]
I wonder whether we should also think about a new prefix `IMPI_` instead of `QMPI_`. I know the `Q` as the "next after P" has a long internal tradition, yet it may not be an intuitive prefix for user. However, may we have potential name conflicts with Intel MPI internal naming?

**Comment by wesbland** [comment]
I think that's a fair point (though it's probably part of https://github.com/mpiwg-tools/mpi-standard/pull/10 😄).

I haven't thought through it that much, but IMO, `FMPI` might make more sense (function pointer interception) and wouldn't cause confusion with `IMPI`. While not part of the standard, it's been around for a long time and has a lot of relatively famous uses of the prefix. It's kinda the same as us not wanting to use `OMPI` to me.

**Comment by devreal** [comment]
I am not sure I like the idea of renaming something that has been in the standard for decades only because there is now an alternative mechanism for achieving the same goal. The term has been used in countless books, publications, and presentations. And renaming the `PMPI` prefix would break far too many codes. Granted, "profiling" was the wrong term to begin with but we have to live with what we have I guess.

If, for some reason, the Forum wants to go down that road I suggest using "interposition" instead of "interception". For one, "library interposition" is an established term in computer science and the `PMPI` interface enables that. Second, it has at least a `P` in it to match the `P` in `PMPI`. And third, it sounds less bellicose (which arguably is a highly subjective argument).

**Comment by schulzm** [comment]
Marc-Andre can correct me, if I am wrong, but I think two things collided here in the discussion: this ticket is intended to clarify the usage of the existing PMPI API, for which profiling has been increasingly confusing due to its usage. Actual profiling is only one of many more cases, as we use the wrapping/interception much more generically these days. Nevertheless, it is of course a term that has been used for very long (even it is incorrect).

However, the ticket does not change the naming - PMPI stays PMPI, as this would indeed break codes. That discussion is about choosing the right prefix for the new interface: the current name QMPI is only a placeholder and not intuitive at all - for this IMPI or FMPI could make sense, but that should probably be taken to a new threads.

**Comment by devreal** [comment]
@schulzm You're right, I misread the discuss about changing prefixes. That discussion should indeed go into a different thread to avoid further confusion (esp since QMPI is not part of MPI yet). The rest of my arguments still stand though :)

**Comment by mahermanns** [comment]
@devreal Yes, no actual API is changed, just how we name and refer to it. However, as @wesbland already noted, the users knowing the term "profiling interface" would have to get used to looking up "interception interface" in the index. Whether that is too big of a hassle, and how we might mitigate such problems should be discussed broadly in the forum, I guess.

The discussion of the new prefix, as you realized, is only about the so-called QMPI interface, which will have a different set of functions to query and register the callback functions for the tools. And the idea here is to come up with an intuitive prefix.

**Comment by wesbland** [comment]
This had a reading on 2022-12-06. There was a lot of feedback about whether or not this change was necessary. I'll mark the ticket as read, but it will likely need a number of changes that would require a re-read.

**Comment by mahermanns** [comment]
We had a straw poll on whether to target this 
a) for MPI 4.1
b) for MPI 5.0
c) not at all
and received 6, 7, and 6 votes respectively. So we defer this to MPI 5.0 and discuss this together with the QMPI interface (yet likely keep the proposed changes separate from QMPI itself)
